guides/contexts.md

```diff
@@ -8,1060 +8,1313 @@
+To close out our order completion, we need to implement the `ShoppingCart.prune_cart_items/1` function in `lib/hello/shopping_cart.ex`:
 
+```elixir
+  def prune_cart_items(%Cart{} = cart) do
+    {_, _} = Repo.delete_all(from(i in CartItem, where: i.cart_id == ^cart.id))
+    {:ok, reload_cart(cart)}
+  end
 ```
-Page created successfully.
-
-Show Page Title: Home
-Body: Welcome to Phoenix!
-Views: 0
-Author: Chris
-```
-
-And it works! We now have two isolated contexts responsible for user accounts and content management. We coupled the content management system to accounts where necessary, while keeping each system isolated wherever possible. This gives us a great base to grow our application from.
 
-## Adding CMS functions
+Our new function accepts the cart struct and issues a `Repo.delete_all` which accepts a query of all items for the provided cart. We return a success result by simply reloading the pruned cart to the caller. With our context complete, we now need to show the user their completed order. Head back to your order controller and modify the `show/2` action:
 
-Just like we extended our `Accounts` context with new application-specific functions like `Accounts.authenticate_by_email_password/2`, let's extend our generated `CMS` context with new functionality. For any CMS system, the ability to track how many times a page has been viewed is essential for popularity ranks. While we could try to use the existing `CMS.update_page` function, along the lines of `CMS.update_page(page, %{views: page.views + 1})`, this would not only be prone to race conditions, but it would also require the caller to know too much about our CMS system. To see why the race condition exists, let's walk through the possible execution of events:
-
-Intuitively, you would assume the following events:
-
-  1. User 1 loads the page with count of 13
-  2. User 1 saves the page with count of 14
-  3. User 2 loads the page with count of 14
-  4. User 2 loads the page with count of 15
-
-While in practice this would happen:
-
-  1. User 1 loads the page with count of 13
-  2. User 2 loads the page with count of 13
-  3. User 1 saves the page with count of 14
-  4. User 2 saves the page with count of 14
-
-The race conditions would make this an unreliable way to update the existing table since multiple callers may be updating out of date view values. There's a better way.
+```elixir
+  def show(conn, %{"id" => id}) do
+-   order = Orders.get_order!(id)
++   order = Orders.get_order!(conn.assigns.current_uuid, id)
+    render(conn, "show.html", order: order)
+  end
+```
 
-Again, let's think of a function name that describes what we want to accomplish.
+We tweaked the show action to pass our `conn.assigns.current_uuid` to `get_order!` which authorizes orders to be viewable only by the owner of the order. Next, we can replace the order show template in `lib/hello_web/templates/orders/show.html.eex`:
 
-    > page = CMS.inc_page_views(page)
+```eex
```

Comment:
Shouldn't it be html instead? The documentation specifies it [here](https://github.com/rubynho/phoenix/blob/master/CONTRIBUTING.md#guides).
```suggestion
```html
```